src/components/organisms/NewResourceWizard/NewResourceWizard.tsx

```diff
@@ -27,12 +31,80 @@ import {openNamespaceTopic, openUniqueObjectNameTopic} from '@utils/shell';
 
 import {getResourceKindHandler} from '@src/kindhandlers';
 
-import {SaveDestinationWrapper, StyledSelect} from './NewResourceWizard.styled';
+import {FileCategoryLabel, FileNameLabel, SaveDestinationWrapper, StyledSelect} from './NewResourceWizard.styled';
 
 const SELECT_OPTION_NONE = '<none>';
 
 const {Option, OptGroup} = Select;
 
+const fetchFullFileName = (filename: string, suffix?: string) => {
+  return `${filename}${suffix || ''}.yaml`;
+};
+
+const createFullFileName = (
+  subfiles: fs.Dirent[],
+  resourceName: K8sResource['name'],
+  resourceKind: K8sResource['kind'],
+  selectedFolder: string,
+  fileMap: FileMapType,
+  suffix: number,
+  includeKind?: boolean
+): string => {
+  const name = resourceName;
+  const nameKind = includeKind ? `-${resourceKind.toLowerCase()}` : '';
+  const nameSuffix = suffix ? ` (${suffix})` : '';
+  const fullFileName = fetchFullFileName(`${name}${nameKind}${nameSuffix}`);
+  let foundFile: fs.Dirent | FileEntry | undefined;
+
+  if (subfiles.length) {
+    foundFile = subfiles.find(dirent => dirent.name === fullFileName);
+  } else if (selectedFolder === ROOT_FILE_ENTRY) {
+    foundFile = fileMap[`${path.sep}${fullFileName}`];
+  } else {
+    foundFile = fileMap[`${path.sep}${path.join(selectedFolder, fullFileName)}`];
+  }
+
+  if (foundFile) {
+    if (includeKind) {
+      return createFullFileName(
+        subfiles,
+        resourceName,
+        resourceKind,
+        selectedFolder,
+        fileMap,
+        suffix ? suffix + 1 : 2,
+        true
+      );
+    }
+    return createFullFileName(
+      subfiles,
+      resourceName,
+      resourceKind,
+      selectedFolder,
+      fileMap,
+      suffix ? suffix + 1 : 0,
+      true
+    );
+  }
+
+  return fullFileName;
+};
+
+const generateFullFileName = async (
+  resourceName: K8sResource['name'],
+  resourceKind: K8sResource['kind'],
+  selectedFolder: string,
+  fileMap: FileMapType
+) => {
+  let subfiles: fs.Dirent[] = [];
+  const fsReaddirPromise = util.promisify(fs.readdir);
+  subfiles = (await fsReaddirPromise(selectedFolder, {withFileTypes: true})).filter(dirent => !dirent.isDirectory());
+
+  const hasNameClash = subfiles.map(x => x.name.split('.')[0]).some(subfile => subfile === resourceName);
```

Comment:
For this we should look through all the resources and find if there is another resource with the same name, similar to how it's done in `generateCreateaAndReplacedFiles` function from `SaveResourcesToFileFolderModal` component

Reply:
This has been amended now. However, I am still wondering if we're only checking name clashes in resource list, the name update will happen at project level, since resources constructed are available in resourceMap. Are we not checking for conflicts inside the selectedFolder?

Reply:
Completely agree, instead of searching name clashes through all the resource names, we should filter from the `resourceMap` only the resources whose `filePath` starts with the `selectedFolder`.